docs/groovy/how-to-guides/keyed-transpose.md

```diff
@@ -0,0 +1,241 @@
+    stringCol("Level", "INFO", "INFO"),
+    intCol("NodeId", 10, 10)
+)
+
+// Define all expected combinations of Level and NodeId
+initGroups = newTable(
+    stringCol("Level", "ERROR", "WARN", "INFO", "ERROR", "WARN", "INFO"),
+    intCol("NodeId", 10, 10, 10, 20, 20, 20)
+).join(source.selectDistinct("Date"))
+
+result = KeyedTranspose.keyedTranspose(
+    source, 
+    List.of(AggCount("Count")), 
+    ColumnName.from("Date"),
+    ColumnName.from("Level", "NodeId"), 
+    initGroups
+)
+```
+
+Even though the source only has INFO logs from NodeId 10, the result will include columns for all Level-NodeId combinations specified in `initGroups`.
+
+## Understanding column naming
+
+The `keyedTranspose` operation follows specific rules for naming output columns:
+
+| Scenario                             | Column Naming Pattern         | Example                  |
+| ------------------------------------ | ----------------------------- | ------------------------ |
+| Single aggregation, single column-by | Value from column-by column   | `INFO`, `WARN`           |
+| Multiple aggregations                | Aggregation name + value      | `Count_INFO`, `Sum_WARN` |
+| Multiple column-by columns           | Values joined with underscore | `INFO_10`, `WARN_20`     |
+| Invalid characters                   | Characters removed            | `1-2.3/4` → `1234`       |
+| Starts with number                   | Prefixed with `column_`       | `123` → `column_123`     |
+| Duplicate names                      | Suffix added                  | `INFO`, `INFO2`          |
+
+### Best practice: Sanitize data before transposing
+
+To maintain control over column names, clean your data values before using `keyedTranspose`:
+
+```groovy order=result,source
+import io.deephaven.engine.table.impl.util.KeyedTranspose
+
+source = newTable(
+    stringCol("Date", "2025-08-05", "2025-08-05", "2025-08-06"),
+    stringCol("Category", "Type-A", "Type-B", "Type-A"),
+    intCol("Value", 10, 20, 15)
+)
+
+// Sanitize category names to be more column-friendly
+cleaned = source.update("CleanCategory = Category.replace('-', '_')")
+
+result = KeyedTranspose.keyedTranspose(
+    cleaned,
+    List.of(AggSum("Total=Value")),
+    ColumnName.from("Date"),
+    ColumnName.from("CleanCategory")
+)
+```
+
+## Common patterns
+
+### Pattern 1: Sales by region and product
+
+```groovy order=salesData,result
+import io.deephaven.engine.table.impl.util.KeyedTranspose
+
+salesData = newTable(
+    stringCol("Product", "Widget", "Widget", "Gadget", "Gadget", "Widget"),
+    stringCol("Region", "North", "South", "North", "South", "East"),
+    intCol("Sales", 100, 150, 200, 175, 125)
+)
+
+// Transform: Product | Region | Sales
+// Into: Product | North | South | East
+result = KeyedTranspose.keyedTranspose(
+    salesData,
+    List.of(AggSum("Sales")),
+    ColumnName.from("Product"),
+    ColumnName.from("Region")
+)
+```
+
+### Pattern 2: Time-series metrics
```

Comment:
IMO presenting these as examples rather than "common patterns" would be beneficial. I don't see how these are common patterns, but they're obviously useful examples for learning.